action/protocol/staking/candidate.go

```diff
@@ -253,7 +253,37 @@ func (l CandidateList) Less(i, j int) bool {
 	if res := l[i].Votes.Cmp(l[j].Votes); res != 0 {
 		return res == 1
 	}
-	return strings.Compare(l[i].Owner.String(), l[j].Owner.String()) == 1
+	switch strings.Compare(l[i].Owner.String(), l[j].Owner.String()) {
```

Comment:
```
func (l CandidateList) Less(i, j int) bool {
	if res := l[i].Votes.Cmp(l[j].Votes); res != 0 {
		return res == 1
	}
	if res := strings.Compare(l[i].Owner.String(), l[j].Owner.String()); res != 0 {
		return res == 1
	}
	if res := strings.Compare(l[i].Operator.String(), l[j].Operator.String()); res != 0 {
		return res == 1
	}
	return strings.Compare(l[i].Name, l[j].Name) == 1
}
```

Reply:
1. @Liuhaai  agree comparing these 2 fields should suffice, and I think since we fix this compare func now, we'd better compare all fields to cover all possible corner cases
2. prefer to use 
```
if res != 0 {
		return res == 1
	}
```
to be consistent, instead of `switch case`